src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/WhitelistApiAction.java

```diff
@@ -0,0 +1,151 @@
+import com.amazon.opendistroforelasticsearch.security.configuration.AdminDNs;
+import com.amazon.opendistroforelasticsearch.security.configuration.ConfigurationRepository;
+import com.amazon.opendistroforelasticsearch.security.dlic.rest.validation.AbstractConfigurationValidator;
+import com.amazon.opendistroforelasticsearch.security.dlic.rest.validation.RolesMappingValidator;
+import com.amazon.opendistroforelasticsearch.security.dlic.rest.validation.WhitelistValidator;
+import com.amazon.opendistroforelasticsearch.security.privileges.PrivilegesEvaluator;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.CType;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableList;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+
+/**
+ * This class implements GET and PUT operations to manage dynamic WhitelistingSettings.
+ * <p>
+ * These APIs are only accessible to SuperAdmin since the configuration controls what APIs are accessible by normal users.
+ * Eg: If whitelisting is enabled, and a specific API like "/_cat/nodes" is not whitelisted, then only the SuperAdmin can use "/_cat/nodes"st
+ * These APIs allow the SuperAdmin to enable/disable whitelisting, and also change the list of whitelisted APIs.
+ * <p>
+ * A SuperAdmin is identified by a certificate which represents a distinguished name(DN).
+ * SuperAdmin DN's can be set in {@link ConfigConstants#OPENDISTRO_SECURITY_AUTHCZ_ADMIN_DN}
+ * SuperAdmin certificate for the default superuser is stored as a kirk.pem file in config folder of elasticsearch
+ * <p>
+ * Example calling the PUT API as SuperAdmin using curl (if http basic auth is on):
+ * curl -v --cacert path_to_config/root-ca.pem --cert path_to_config/kirk.pem --key path_to_config/kirk-key.pem -XPUT https://localhost:9200/_opendistro/_security/api/whitelist -H "Content-Type: application/json" -d’
+ * {
+ * "whitelistingEnabled" : false,
+ * "whitelistedAPIs" : ["/_cat/nodes","/_opendistro/_security/api/whitelist","/_opendistro/_security/api/securityconfig"]
+ * }
+ * ‘
+ * <p>
+ * Currently, whitelisting checks the path for equality, so make sure you don't have errors in the whitelisted APIs.
```

Comment:
Will definitely add a comment in the yml file for the expected format. I had avoided handling the extra '/', because I thought it would increase the time complexity, since I'd have to iterate through each new element in the set and remove extra '/' from the API name.

Reply:
Fixed this, added additional logic to allow extra '/'.